### PR TITLE
feat: add documentation

### PR DESCRIPTION
**Pull request by hansieodendaal**
- Updated the `README.md` with some pertinent development and benchmark information
- Added a missing license identifier
- Fixed clippy lint errors

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,90 @@
...
 
-Based on the proof of concept Python implementation for Bulletproofs+ at https://github.com/AaronFeickert/pybullet-plus 
-by [Aaron Feickert](https://github.com/AaronFeickert). Bulletproofs+ was originally intended for 
-[Monero](https://www.getmonero.org/2020/12/24/Bulletproofs+-in-Monero.html). 
\ No newline at end of file
+## Overview
+
+Tari Bulletproofs+ is an implementation of _Bulletproofs+: Shorter Proofs for Privacy-Enhanced Distributed Ledger_ [2], 
+which was derived from the original _Bulletproofs: Short Proofs for Confidential Transactions and More_ [1]. The former 
+offer a 96 bytes shorter proof size than the latter.
```
Comment (SWvheerden) [inline README.md:9]:
```suggestion
offers a 96 bytes shorter proof size than the latter.
```

File: README.md [hunk 2]
```diff
@@ -2,6 +2,90 @@
...
+
+## Comparative performance
+
+As we intend moving from Bulletproofs [1] to Bulletproofs+ [2] in our
```
Comment (SWvheerden) [inline README.md:13]:
```suggestion
As we intend to move from Bulletproofs [1] to Bulletproofs+ [2] in our 
```

File: README.md [hunk 3]
```diff
@@ -2,6 +2,90 @@
...
+
+Tari Bulletproofs+ is an implementation of _Bulletproofs+: Shorter Proofs for Privacy-Enhanced Distributed Ledger_ [2], 
+which was derived from the original _Bulletproofs: Short Proofs for Confidential Transactions and More_ [1]. The former 
+offer a 96 bytes shorter proof size than the latter.
+
+## Comparative performance
+
+As we intend moving from Bulletproofs [1] to Bulletproofs+ [2] in our 
+[blockchain poject](https://github.com/tari-project), the natural benchmark comparison is with Dalek's Bulletproofs [4]. 
+On average our proof creation and proof verification is respectively 32% and 4% slower, however, employing batch
```
Comment (SWvheerden) [inline README.md:15]:
```suggestion
On average, our proof creation and proof verification are respectively 32% and 4% slower. However, employing batch 
```

File: README.md [hunk 4]
```diff
@@ -2,6 +2,90 @@
...
+Tari Bulletproofs+ is an implementation of _Bulletproofs+: Shorter Proofs for Privacy-Enhanced Distributed Ledger_ [2], 
+which was derived from the original _Bulletproofs: Short Proofs for Confidential Transactions and More_ [1]. The former 
+offer a 96 bytes shorter proof size than the latter.
+
+## Comparative performance
+
+As we intend moving from Bulletproofs [1] to Bulletproofs+ [2] in our 
+[blockchain poject](https://github.com/tari-project), the natural benchmark comparison is with Dalek's Bulletproofs [4]. 
+On average our proof creation and proof verification is respectively 32% and 4% slower, however, employing batch 
+verification, immediate benefits are evident with gains ranging from 38% to 74% for batch sizes from 2 to 32 proofs.
```
Comment (SWvheerden) [inline README.md:16]:
```suggestion
verification, immediate benefits are evident, with gains ranging from 38% to 74% for batch sizes from 2 to 32 proofs. 
```

File: README.md [hunk 5]
```diff
@@ -2,6 +2,90 @@
...
+offer a 96 bytes shorter proof size than the latter.
+
+## Comparative performance
+
+As we intend moving from Bulletproofs [1] to Bulletproofs+ [2] in our 
+[blockchain poject](https://github.com/tari-project), the natural benchmark comparison is with Dalek's Bulletproofs [4]. 
+On average our proof creation and proof verification is respectively 32% and 4% slower, however, employing batch 
+verification, immediate benefits are evident with gains ranging from 38% to 74% for batch sizes from 2 to 32 proofs. 
+Thus, with our current implementation, we could not recreate the experimental results in [2], which claimed 16% 
+reduction in prover time, however, our 4% increase in verification time is exactly on par with their 3%.
```
Comment (SWvheerden) [inline README.md:18]:
```suggestion
reduction in prover time. However, our 4% increase in verification time is exactly on par with their 3%. 
```

**Review by SWvheerden — APPROVED** [review]
The looks good to me. 
I run the docs through Grammarly, those changes are all Grammarly. 

I would have preferred to have 2 PR's here one for Docs, one for Clippy fixes, or at least have them split into two separate commits. But luckily all changes where still small.

**Comment by hansieodendaal** [comment]
> The looks good to me. I run the docs through Grammarly, those changes are all Grammarly.
> 
> I would have preferred to have 2 PR's here one for Docs, one for Clippy fixes, or at least have them split into two separate commits. But luckily all changes where still small.

Yes, agreed and nice touch verifying with Grammarly, thanks.

**Review by mikethetike — APPROVED** [review]

**Pull request merged**